app/src/main/java/edu/uw/ischool/opensecrets/SecretApp.kt

```diff
@@ -1,17 +1,171 @@
 import edu.uw.ischool.opensecrets.util.OptionManager
 
 
+const val CHANNEL_ID = "OpenSecretNotifications"
 class SecretApp : Application() {
+
+    private lateinit var journal: File
+    private val contactMessager : ContactMessager = ContactMessager()
+    lateinit var messageQueue : TimedMessageQueue
+    lateinit var alarmManager: AlarmManager
+    private var messageID = 1
+    private val sampleEntry : Entry = Entry("My Title", "blue", "Call me Ishmael. Some years ago...", Date())
     lateinit var journalManager: JournalManager
     lateinit var optionManager: OptionManager
     override fun onCreate() {
         journalManager = JournalManager(this)
         optionManager = OptionManager(this)
         super.onCreate()
+        createNotificationChannel()
+        alarmManager = getSystemService(Context.ALARM_SERVICE) as AlarmManager
+        messageQueue = TimedMessageQueue(this, alarmManager, ::sendEntryToRandomContact)
+        updateContactList()
+//        sendNotification("Notification title", "Hello there!")
+//        queueSendEntry(sampleEntry)
+    }
+
+    
+
+    fun journalExist(): Boolean {
+        return journal.exists()
+    }
+
+    fun appendEntry(entry: Entry): Boolean {
+        // check if journal exist
+        if (journalExist()) {
+            // load og data
+            val inputStream = FileReader(journal)
+            var data: JSONArray
+            inputStream.use {
+                data = try {
+                    JSONArray(it.readText())
+
+                } catch (e: JSONException) {
+                    JSONArray()
+                }
+            }
+            inputStream.close()
+            // create new data obj
+            val jsonEntry = JSONObject()
+            jsonEntry.put("title", entry.title)
+            jsonEntry.put("text", entry.text)
+            jsonEntry.put("color", entry.color)
+            jsonEntry.put("dateCreated", entry.dateCreated.toString())
+            data.put(jsonEntry)
+            // create new file directory
+            val fileOutput = File(filesDir, "new_journal.json")
+            fileOutput.createNewFile()
+            // write to new file
+            val outputStream = FileWriter(fileOutput)
+            outputStream.use {
+                it.write(data.toString())
+                it.flush()
+            }
+            outputStream.close()
+            // check file data
+            val newData = FileReader(fileOutput).use {
+                JSONArray(it.readText())
+            }
+            if (verifyJSON(data, newData)) {
+                journal.delete()
+                fileOutput.renameTo(File(filesDir, "journal.json"))
+            } else {
+                Toast.makeText(this, "data not the same", Toast.LENGTH_SHORT).show()
+            }
+            return true
+        } else {
+            return false
+        }
     }
 
+    private fun verifyJSON(ogData: JSONArray, newData: JSONArray): Boolean {
```

Comment:
remove this since JournalManager has been implemented, so call JournalManager instead for this.